### PR TITLE
And multi-path in RowsetGraph

### PR DESCRIPTION
**Pull request by chaoyli**
1. Do not remove rowset from RowsetGraph after compaction
2. Collect unused rowset at interval
3. Remove inc_version and related code

### CODE REVIEW

**Review by yiguolei — CHANGES_REQUESTED** [review]

File: be/src/olap/compaction.cpp [hunk 1]
```diff
@@ -110,44 +110,22 @@ OLAPStatus Compaction::construct_input_rowset_readers() {
...
 }
 
-OLAPStatus Compaction::modify_rowsets() {
+OLAPStatus Compaction::add_rowsets() {
     std::vector<RowsetSharedPtr> output_rowsets;
     output_rowsets.push_back(_output_rowset);
     
-    WriteLock wrlock(_tablet->get_header_lock_ptr());
-    OLAPStatus res = _tablet->modify_rowsets(output_rowsets, _input_rowsets);
+    OLAPStatus res = _tablet->add_rowsets(output_rowsets);
```
[the remark this answers is not in the document]
Comment (chaoyli) [reply be/src/olap/compaction.cpp]: I will call add_rowset() instead.

File: be/src/olap/task/engine_clone_task.cpp [hunk 2]
```diff
@@ -760,15 +775,14 @@ OLAPStatus EngineCloneTask::_clone_full_data(TabletSharedPtr tablet, TabletMeta*
...
         Version local_version(rs_meta->start_version(), rs_meta->end_version());
         VersionHash local_version_hash = rs_meta->version_hash();
         LOG(INFO) << "check local delta when full clone."
-            << "tablet=" << tablet->full_name()
-            << ", local_version=" << local_version.first << "-" << local_version.second;
+                  << "tablet=" << tablet->full_name()
+                  << ", local_version=" << local_version.first << "-" << local_version.second;
 
         // if local version cross src latest, clone failed
         // if local version is : 0-0, 1-1, 2-10, 12-14, 15-15,16-16
```
[the remark this answers is not in the document]
Comment (chaoyli) [reply be/src/olap/task/engine_clone_task.cpp]: Check cloned_tablet_meta is complete should construct the RowsetGraph.
But RowsetGraph is only visible in Tablet.
So, checking destination rowsets is complete or not by add_rowsets is enough

File: be/src/olap/task/engine_clone_task.cpp [hunk 3]
```diff
@@ -811,26 +824,22 @@ OLAPStatus EngineCloneTask::_clone_full_data(TabletSharedPtr tablet, TabletMeta*
                         << ", version='" << local_version.first<< "-" << local_version.second
                         << ", version_hash=" << local_version_hash;
                 }
-            } else {
```
[the remark this answers is not in the document]
Comment (chaoyli) [reply be/src/olap/task/engine_clone_task.cpp]: Version of these rowsets are the same with rowsets in RowsetGraph.
Current logic do not take rowsetid into consideration.
So correct rowset may be remove mistakenly.

File: be/src/olap/task/engine_storage_migration_task.cpp [hunk 4]
```diff
@@ -246,7 +246,8 @@ OLAPStatus EngineStorageMigrationTask::_generate_new_header(
     for (auto& rs : consistent_rowsets) {
         rs_metas.push_back(rs->rowset_meta());
     }
-    new_tablet_meta->revise_rs_metas(rs_metas);
```
[the remark this answers is not in the document]
Comment (chaoyli) [reply be/src/olap/task/engine_storage_migration_task.cpp]: I found reset(new TabletMeta()) will also clear TabletSchema.
I add a interface name clean_and_add_rs_metas() to only clean old rs_metas and add new rs_metas.

File: be/src/olap/tablet_manager.cpp [hunk 5]
```diff
@@ -1424,4 +1418,13 @@ TabletSharedPtr TabletManager::_get_tablet_with_no_lock(TTabletId tablet_id, Sch
     return tablet;
 } // _get_tablet_with_no_lock
 
+void TabletManager::capture_unused_rowset_in_tablet() {
+    ReadLock rdlock(&_tablet_map_lock);
```
[the remark this answers is not in the document]
Comment (chaoyli) [reply be/src/olap/tablet_manager.cpp]: I will put tablet into a tablet_vec

**Pull request closed without merging**